lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -260,26 +306,88 @@ public static Domain getDomain(
         return Domain.create(ValueSet.all(type), hasNullValue);
     }
 
-    private static Optional<ParquetIntegerStatistics> toParquetIntegerStatistics(Statistics<?> statistics, ParquetDataSourceId id, String column)
+    @VisibleForTesting
+    public Domain getDomain(Type type, ColumnIndex columnIndex, ParquetDataSourceId id, RichColumnDescriptor descriptor)
             throws ParquetCorruptionException
     {
-        if (statistics instanceof LongStatistics) {
-            LongStatistics longStatistics = (LongStatistics) statistics;
-            if (longStatistics.genericGetMin() > longStatistics.genericGetMax()) {
```

Comment:
This check seems to be gone. Why?

Reply:
This is checked when the range is created deeper in the code.